src/filters/method.rs

```diff
@@ -137,3 +114,69 @@ where
     })
 }
 
+pub mod v2 {
+    //! HTTP Method Filters
+    //!
+    //! These filters deal with the HTTP Method part of a request. They match
+    //! the request `Method`, and if not matched, will reject the request with a
+    //! `405 Method Not Allowed`.
+    use http::Method;
+
+    use filter::Filter;
+    use reject::Rejection;
+
+    use super::method_is;
```

Comment:
while here, could you add a short comment on `method_is` above explaining why it takes an Fn? copying https://github.com/seanmonstar/warp/pull/21#discussion_r207746190 would be enough.